integration/mediation-tests/tests-transport/src/test/java/org/wso2/carbon/esb/ssl/test/DynamicSSLProfilesPTTListenerTestCase.java

```diff
@@ -0,0 +1,177 @@
+    private String keyStorePassword = "wso2carbon";
+    private String trustStorePassword = "wso2carbon";
+    private String proxyService = "sslTestProxy";
+    private String serviceName =
+            "org.apache.synapse:Type=ListenerSSLProfileReloader," + "Name=PassThroughHttpMultiSSLListener";
+    private String configLocation =
+            getESBResourceLocation() + File.separator + "dynamicsslprofiles" + File.separator + "pttlistener"
+                    + File.separator;
+
+    @SetEnvironment(executionEnvironments = { ExecutionEnvironment.STANDALONE })
+    @BeforeClass(alwaysRun = true)
+    public void initialize() throws Exception {
+        super.init();
+        serverManager = new ServerConfigurationManager(context);
+        serverManager.applyConfigurationWithoutRestart(new File(
+                getESBResourceLocation() + File.separator + "dynamicsslprofiles" + File.separator + "pttlistener"
+                        + File.separator + "axis2.xml"));
+        serverManager.restartGracefully();
+        super.init();
+    }
+
+    @SetEnvironment(executionEnvironments = { ExecutionEnvironment.STANDALONE })
+    @Test(groups = { "wso2.esb" },
+          description = "Testing Mutual SSL Connection with dynamically loaded SSL configuration")
+    public void testMutualSSLConnectionWithUpdatedProfile() throws Exception {
+        String soapMessage = "<soapenv:Envelope "
+                + "xmlns:soapenv=\"http://schemas.xmlsoap.org/soap/envelope/\" "
+                + "xmlns:echo=\"http://echo.services.core.carbon.wso2.org\">\n"
+                + "   <soapenv:Header/>\n"
+                + "   <soapenv:Body>\n"
+                + "      <echo:echoString>\n"
+                + "         <in>WSO2 Stock</in>\n"
+                + "      </echo:echoString>\n"
+                + "   </soapenv:Body>\n" + "</soapenv:Envelope>";
+
+        //load key store file
+        MutualSSLClient.loadKeyStore(configLocation + keyStoreName, keyStorePassword);
+        //load trust store file
+        MutualSSLClient.loadTrustStore(configLocation + trustStoreName, trustStorePassword);
+        //create ssl socket factory instance with given key/trust stores
+        MutualSSLClient.initMutualSSLConnection();
+
+        updateProfileConfigurationFiles();
+        invokeJMXMethod();
+
+        Map<String, String> reqProps = new HashMap<String, String>();
+        reqProps.put("Content-type", "text/xml; charset=utf-8");
+        reqProps.put("SOAPAction", "urn:echoString");
+
+        String response;
+        try {
+            response = MutualSSLClient.sendPostRequest(getProxyServiceURLHttps(proxyService), soapMessage, reqProps);
+        } catch (Exception e) {
+            log.info("Error sending Post request to proxy service", e);
+            response = "";
+        }
+
+        rollbackProfileConfigurationFiles();
+        assertTrue("Mutual SSL Error because of incorrect key", response.contains("WSO2 Stock"));
```

Comment:
As I remember in `Assert.assertTrue()`, 1st parameter is the boolean condition and 2nd parameter is the message. I am wondering how this work.

Reply:
assertTrue method used here is from Junit. For TestNG's assertTrue only, 1st param is condition and then the message.